### PR TITLE
fix: identifiers are not incorrectly scoped to state

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Description

https://builder-io.atlassian.net/browse/ENG-7338

Mitosis transforms identifiers that have the same name as state variables to have `this.` or `state.` prefixed where necessary. This causes issues with shadowed variables that are unrelated to state.

In the following example, the `errors` in `foo` have nothing to do with the `errors` defined in state.

```js
const state = useStore({
  errors: {},
  foo(errors) {
    return errors;
  }
});
```

However, this is incorrectly prefixed with `state.` which causes runtime issues:

```js
const state = useStore({
  errors: {},
  foo(state.errors) {
    return state.errors;
  }
});
```

This transformation is only needed when moving functions into state or when using `useState`. To fix this, I updated Mitosis to only transform state values that were set using `useState` or were function that got moved to state.

Make sure to follow the PR preparation steps in [CONTRIBUTING.md](../CONTRIBUTING.md#preparing-your-pr) before submitting your PR:

- [x] format the codebase: from the root, run `yarn fmt:prettier`.
- [x] update all snapshots (in core & CLI): from the root, run `yarn test:update`
- [x] add Changeset entry: from the root, run `yarn g:changeset` and follow the CLI instructions. Alternatively, use the Changeset Github Bot to create the file.

### CODE REVIEW

File: .changeset/late-papayas-eat.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@builder.io/mitosis': patch
+---
+
+identifiers are not incorrectly scoped to state
```
Comment (samijaber) [inline .changeset/late-papayas-eat.md:5]: can we follow the changeset format here? what exactly got fixed, for which generators/parsers, etc.

Comment (liamdebeasi) [reply .changeset/late-papayas-eat.md]: updated in https://github.com/BuilderIO/mitosis/pull/1630/commits/ed6b5f81b5879abb525f3700e377c09bc5503472

Let me know if you have other ideas for how to word this

File: .changeset/late-papayas-eat.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@builder.io/mitosis': patch
+---
+
+shadowed variables in state methods are not scoped to state in all generators
```
Comment (samijaber) [inline .changeset/late-papayas-eat.md:5]: [All] Fix: scope renaming of state methods to not include shadow variables
[Angular]: Update `state.*` -> `this.*` transform to new AST transform approach

Comment (samijaber) [inline .changeset/late-papayas-eat.md:5]:
```suggestion
[All] Fix: scope renaming of state methods to not include shadow variables
[Angular]: Update `state.*` -> `this.*` transform to new AST transform approach
```

**Review by samijaber — APPROVED** [review]

**Pull request merged**